src/sip/request.c

```diff
@@ -176,15 +184,39 @@ static int request(struct sip_request *req, enum sip_transp tp,
 
 	(void)re_snprintf(branch, 24, "z9hG4bK%016llx", rand_u64());
 
+	/* get laddr corresponding to destination */
 	err = sip_transp_laddr(req->sip, &laddr, tp, dst);
 	if (err)
 		goto out;
 
+	/* add request line and Via header */
 	err  = mbuf_printf(mb, "%s %s SIP/2.0\r\n", req->met, req->uri);
 	err |= mbuf_printf(mb, "Via: SIP/2.0/%s %J;branch=%s;rport\r\n",
 			   sip_transp_name(tp), &laddr, branch);
+
+	/* add Contact header */
 	err |= req->sendh ? req->sendh(tp, &laddr, dst, mb, req->arg) : 0;
 	err |= mbuf_write_mem(mb, mbuf_buf(req->mb), mbuf_get_left(req->mb));
+
+	/* add Content-Length and SDP message body (if present) */
+	sess = req->sdp;
+	if (sess) {
+		sdp_session_set_laddr(sess, &laddr);
+		bb = mbuf_alloc(1024);
```

Comment:
I think this is not needed since `sdp_encode` allocates mbuf.

Reply:
Should the description of `sdp_encode` be fixed?
```
/**
 * Encode an SDP Session into a memory buffer
 *
 * @param mbp   Pointer to allocated memory buffer
 *
```
Based on that I thought that the user needs to allocate the buffer.